module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -66,8 +68,29 @@ public function listAction()
             return $view->cancelResults;
         }
 
-        // By default, assume we will not need to display a form:
+        // Process any update request results stored in the session:
+        $updateResultsContainer = $this->getHoldUpdateResultsContainer();
+        $holdUpdateResults = $updateResultsContainer->results ?? null;
+        if ($holdUpdateResults) {
+            $view->updateResults = $holdUpdateResults;
+            $updateResultsContainer->results = null;
```

Comment:
Should this logic come AFTER the POST-processing logic? Otherwise, it seems there might be some edge case somewhere where we consume results from the container, but then never display them because the update code forwarded somewhere else.

Reply:
(Or could we make the whole thing more clear by actually creating a separate action to be the target of the form, so we don't need to have so many conditional paths inside this one method?).

Reply:
A separate path is compelling, but it makes things difficult if the form needs to be displayed again due to validation issues. But perhaps it makes sense to branch to sub-methods internally.

There should be no such case where we have update results but are still redirected to the update form. If that were to happen, it would still make sense to clear the old results as otherwise displaying them later might be confusing as they wouldn't necessarily relate to the changes submitted.

Reply:
That makes sense. I do agree that branching to sub-methods might be a good compromise solution. In the MyResearchController, there were too many methods, so we made the methods very long to keep related things together. Now that we have a whole controller for holds, I think we can get away with spreading out to more methods for the sake of improved readability and maintenance.

Reply:
I think it's now better already with your proposed changes of splitting blocks out of editAction. What remains is mostly just about displaying the form and results. What do you think?